src/transfer_zip.py

```diff
@@ -0,0 +1,657 @@
+    )
+
+    parser.add_argument(
+        "--dry_run",
+        action="store_true",
+        help="Do not transfer any data or metadata; only log.",
+    )
+
+    parser.add_argument(
+        "--window",
+        type=str,
+        required=False,
+        help=(
+            "Time to search past the embargo time period."
+            " Specified in astropy quantity_str format (e.g. '3hr 5min 20s')."
+            " Defines a window of data that will be transferred."
+            " This is helpful for transferring a limited amount of data"
+            " during the initial testing of the deployment."
+        ),
+    )
+    parser.add_argument(
+        "--now",
+        default=None,
+        type=str,
+        help=(
+            "Now time in (ISOT, TAI timescale)."
+            " If left blank it will use astropy.time.Time.now."
+        ),
+    )
+
+    parser.add_argument(
+        "-C",
+        "--config_file",
+        required=True,
+        type=str,
+        help="Path to YAML config file.",
+    )
+
+    parser.add_argument(
+        "-d",
+        "--dest_uri_prefix",
+        type=str,
+        default="/sdf/data/rubin/lsstdata/offline/instrument/",
+        help="Destination URI prefix for raw data.",
+    )
+
+    parser.add_argument(
+        "-r",
+        "--rucio_rse",
+        type=str,
+        required=False,
+        help="Rucio RSE for raw data.",
+    )
+    parser.add_argument(
+        "-s",
+        "--scope",
+        type=str,
+        required=False,
+        help="Rucio scope for raw data.",
+    )
+
+    parser.add_argument(
+        "--log",
+        type=str,
+        required=False,
+        help="Level name (WARNING, INFO, DEBUG) or comma-sepaarated list of logger=level pairs.",
+    )
+
+    ns = parser.parse_args()
+    ns.now = Time(ns.now, format="isot", scale="tai") if ns.now else Time.now()
+    if ns.now > Time.now():
+        raise ValueError(f"--now is in the future: {ns.now}")
+    if ns.rucio_rse is not None:
+        if ns.scope is None:
+            raise ValueError("--scope required with --rucio_rse")
+
+    return ns
+
+
+def transfer_data_query(data_query: DataQuery) -> None:
+    """Transfer all files matching a data query into zip files.
+
+    Parameters
+    ----------
+    data_query: `DataQuery`
+        The query and associated embargo time.
+    """
+    global logger, config, source_butler
+
+    # End of window is now - embargo length
+    end_time = config.now - TimeDelta(data_query.embargo_hours * 3600, format="sec")
+    # If window is defined, then start is that much before the end
+    # Otherwise, start is infinitely previous
+    if config.window is not None:
+        start_time = end_time - TimeDelta(config.window, format="quantity_str")
+    else:
+        start_time = Time(0, format="jd")
+    ok_timespan = Timespan(start_time, end_time)
+
+    # Find all exposures meeting criteria
+    dim_where = "(exposure.timespan OVERLAPS _ok_timespan)"
```

Comment:
Going forward we are advocating the new explicit bind syntax or `:_ok_timespan`.

Reply:
I wouldn't need the initial underscore if there's a colon.  I didn't see this new feature/deprecation; I guess I need to follow the release notes better.